src/push_service/keys.rs

```diff
@@ -83,13 +87,13 @@ impl PushService {
 
     pub(crate) async fn get_pre_keys(
         &mut self,
-        destination: &ServiceAddress,
+        destination: &ServiceId,
         device_id: u32,
     ) -> Result<Vec<PreKeyBundle>, ServiceError> {
         let path = if device_id == 1 {
-            format!("/v2/keys/{}/*?pq=true", destination.uuid)
+            format!("/v2/keys/{}/*?pq=true", destination.raw_uuid())
         } else {
-            format!("/v2/keys/{}/{}?pq=true", destination.uuid, device_id)
+            format!("/v2/keys/{}/{}?pq=true", destination.raw_uuid(), device_id)
         };
```

Comment:
Owkay we have duplication of the pre key path now, maybe we want this deduped? Especially with the destination logic in there :-)

Reply:
How about dealing with those in another PR instead?

Reply:
Sure!